pallets/merkle/src/lib.rs

```diff
@@ -571,6 +601,45 @@ impl<T: Config> Tree<T> for Pallet<T> {
 		Ok(tree_id)
 	}
 
+	fn initialize_tree(tree_id: T::TreeId, key_id: T::KeyId) -> Result<(), DispatchError> {
+		let mut tree = Trees::<T>::get(tree_id).ok_or(Error::<T>::TreeDoesntExist)?;
+		ensure!(!tree.initialized, Error::<T>::AlreadyInitialized);
+		let params = Self::get_verifying_key(key_id)?;
+		let (zero_tree, root_hash) = tree.setup.generate_zero_tree::<T>(tree.depth as usize, &params)?;
+		tree.root_hash = Some(root_hash);
+		tree.edge_nodes = Some(zero_tree.clone());
+		tree.zero_tree = Some(zero_tree);
```

Comment:
Do these need to be optional, since we have an `initialized` flag?
Looks like it will panic here: https://github.com/webb-tools/anon/pull/157/files#diff-f1454fc09c24f69884b2f3befb9dd62814169dc43b3ee8e63a707f1245ff0960R817 if there are not present.
Why not just check the `initialized` flag when in `add_members` and leave these as they were.

Or even further, we could separate verifying keys from the tree and check if the key exists based on the setup. Then there is no need for the initialization step.

Or there is something I don't see yet (my 1h review time is up)

Reply:
I suppose not, will rework it and make new trees default to empty vecs